client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js

```diff
@@ -129,8 +129,11 @@ export default function WPCheckout( {
 	] = useState( false );
 
 	const validateContactDetailsAndDisplayErrors = async () => {
-		debug( 'validating contact details with side effects' );
-		if ( areThereDomainProductsInCart ) {
+		debug( 'validating contact details and reporting errors' );
+		const areDomainDetailsNeeded = needsDomainDetails( responseCart ) || isGSuiteInCart;
```

Comment:
I guess you could move declaring `areDomainDetailsNeeded` to the scope above and then share it between `validateContactDetailsAndDisplayErrors` and `validateContactDetails`?

Reply:
Hm... you're right. Although the difference isn't just scope; it's when the function is being called. Inside this block, it will be called on validation. Inside the outer block, it will be called on every render. It's a small calculation so it's probably ok, but let me think about it...

Reply:
Yeah, this makes sense. Pushed a change.